starlite/app.py

```diff
@@ -438,15 +442,99 @@ def handler() -> None:
 
             handler_index = app.get_handler_index_by_name("my-handler")
 
-            # { "path": "/", "handler" ... }
+            # { "path": ["/"], "handler" ... }
             ```
         Args:
             name: A route handler unique name.
 
         Returns:
             A [HandlerIndex][starlite.app.HandlerIndex] instance or None.
         """
-        return self._route_handler_index.get(name)
+        handler = self._route_handler_index.get(name)
+        if not handler:
+            return None
+
+        qualname = self._get_handler_qualname(handler)
+
+        routes = self._route_mapping[qualname]
+        paths = sorted(unique([route.path for route in routes]))
+
+        return HandlerIndex(handler=handler, paths=paths, qualname=qualname)
+
+    def route_reverse(self, name: str, **path_parameters: Any) -> Optional[str]:
+        """Receives a route handler name, path parameter values and returns an
+        optional url path to the handler with filled path parameters.
+
+        Examples:
+            ```python
+            from starlite import Starlite, get
+
+
+            @get("/group/{group_id:int}/user/{user_id:int}", name="get_memebership_details")
+            def get_membership_details(group_id: int, user_id: int) -> None:
+                pass
+
+
+            app = Starlite(route_handlers=[get_membership_details])
+
+            path = app.route_reverse("get_memebership_details", user_id=100, group_id=10)
+
+            # /group/10/user/100
+            ```
+        Args:
+            name: A route handler unique name.
+            **path_parameters: Actual values for path parameters in the route.
+
+        Raises:
+            ValidationException: If path parameters are missing in **path_parameters or have wrong type.
+
+        Returns:
+            A fully formatted url path or None.
+        """
+        handler_index = self.get_handler_index_by_name(name)
+        if handler_index is None:
+            return None
+
+        allow_str_instead = [datetime, date, time, timedelta, float, Path]
```

Comment:
not sure it will work, but better to use a set for this because it way faster.

Reply:
It works with set just fine